### PR TITLE
Forgets device if we fail to get accounts

### PR DESCRIPTION
**Pull request by douglance**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -123,7 +123,7 @@ class LatticeKeyring extends EventEmitter {
   // Add addresses to the local store and return the full result
   async addAccounts(n=1) {
     if (n === CLOSE_CODE) {
```
Comment (alex-miller-0) [inline index.js:125]: You can remove this entire block. It's a remnant from a previous design

Comment (alex-miller-0) [reply index.js:125]: Separately, you can also remove this block as it is quite old and no longer necessary: https://github.com/GridPlus/eth-lattice-keyring/blob/master/index.js#L93

File: index.js [hunk 2]
```diff
@@ -578,7 +578,7 @@ class LatticeKeyring extends EventEmitter {
       // the device is unplugged it will time out and we don't need to wait
       // 2 minutes for that to happen.
       this.sdkSession.timeout = CONNECT_TIMEOUT;
-      await this.sdkSession.connect(this.creds.deviceID)
+      return this.sdkSession.connect(this.creds.deviceID)
```
Comment (alex-miller-0) [inline index.js:581]: Why are you returning this now? Does that mean `finally` won't get hit?

Comment (douglance) [reply index.js:581]: If I return the promise, then I can get the `isPaired` return value without having to set a variable here and then return that. It also lets us handle the `.connect` call in the function that calls `_connect` so we could chain `.then()` or `.catch()` calls after `_connect` now.   
  
`finally` always fires.   
  
For both questions, I threw together a minimal example of the same code paths here: [https://codesandbox.io/s/charming-pateu-eldrqs?file=/src/index.js](https://codesandbox.io/s/charming-pateu-eldrqs?file=/src/index.js) If you look at the CodeSandbox console when this runs, you see that the finally block is always logged. Then if you comment out line 16 and uncomment line 17, you can see how we can pass around the `Promise` to catch errors where we want.

File: index.js (hunk 1 above)
Comment (douglance) [reply index.js:125]: removed both 9fb5aa5

**Pull request merged**